x/sqlbuilder/select_statement.go

```diff
@@ -0,0 +1,118 @@
+package sqlbuilder
+
+import (
+	"fmt"
+	"io"
+	"strings"
+)
+
+type SelectStatement struct {
+	Table string
+
+	JoinClauses   []JoinClause
+	SelectClauses []Marker
+	WhereClause   PredicateClause
+	GroupByClause []Marker
+	HavingClause  PredicateClause
+}
+
+type JoinType string
+
+const (
+	DefaultJoin JoinType = ""
```

Comment:
What is the use of this ?

Reply:
JOIN with no INNER nor OUTER  token in front of it, is a valid ANSI SQL expression. therefore i tried to represent it here, may be useless? on tje DB systems with use it is equivalent to a INNER JOIN